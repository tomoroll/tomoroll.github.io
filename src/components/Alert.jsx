import React from 'react';
import { string, node, oneOfType } from 'prop-types';

function Alert({ classType, message }) {
  if (!message) {
    return null;
  }
  return (
    <div className={`alert alert-${classType}`} role="alert">
      {message}
    </div>
  );
}
Alert.propTypes = {
  classType: string,
  message: oneOfType([string, node]),
};
Alert.defaultProps = {
  classType: 'primary',
  message: null,
};

export default Alert;
