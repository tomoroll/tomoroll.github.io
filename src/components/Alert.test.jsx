import React from 'react';
import renderer from 'react-test-renderer';
import Alert from './Alert';

it('renders correctly message', () => {
  const tree = renderer
    .create(<Alert classType="success" message="message" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly no message', () => {
  const tree = renderer
    .create(<Alert classType="success" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
