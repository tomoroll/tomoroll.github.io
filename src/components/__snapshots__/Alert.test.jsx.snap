// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly message 1`] = `
<div
  className="alert alert-success"
  role="alert"
>
  message
</div>
`;

exports[`renders correctly no message 1`] = `null`;
